File: packages/react-native-messaging-core/ios/Sources/Models/ConversationItems.swift
import Foundation
import NablaCore
import NablaMessagingCore

extension PaginatedList where Element == ConversationItem {
    var dictionaryRepresentation: [String: Any] {
        var result = [String: Any]()
        result["hasMore"] = hasMore
        result["items"] = elements.map(\.dictionaryRepresentation)
        return result
    }
}

File: packages/react-native-messaging-core/src/types/ConversationItems.ts
import { ConversationItem } from './ConversationItem';

export class ConversationItems {
  hasMore: boolean;
  items: ConversationItem[];

  constructor(hasMore: boolean, items: ConversationItem[]) {
    this.hasMore = hasMore;
    this.items = items;
  }
}
